File: frontend/frontend/src/types/analysis.ts
export interface GitHubProfile {
  login: string;
  name: string | null;
  avatar_url: string;
  bio: string | null;
  html_url: string;
  followers: number;
  following: number;
  public_repos: number;
}

export interface RepositorySummary {
  name: string;
  full_name: string;
  html_url: string;
  description: string | null;
  language: string | null;
  stars: number;
  forks: number;
}

export interface LanguageStat {
  language: string;
  count: number;
  percentage: number;
}

export interface Statistics {
  total_repos: number;
  followers: number;
  following: number;
  total_stars: number;
  total_forks: number;
  languages: LanguageStat[];
  top_repositories: RepositorySummary[];
}

export interface Analysis {
  username: string;
  github_profile: GitHubProfile;
  repositories: RepositorySummary[];
  statistics: Statistics;
  roast: string;
  created_at: string;
}

export interface ApiErrorBody {
  detail: string | { msg: string; type: string }[];
}

File: backend/app/schemas/analysis.py
from datetime import datetime

from pydantic import BaseModel


class GitHubProfile(BaseModel):
    login: str
    name: str | None = None
    avatar_url: str
    bio: str | None = None
    html_url: str
    followers: int
    following: int
    public_repos: int


class RepositorySummary(BaseModel):
    name: str
    full_name: str
    html_url: str
    description: str | None = None
    language: str | None = None
    stars: int
    forks: int


class LanguageStat(BaseModel):
    language: str
    count: int
    percentage: float


class Statistics(BaseModel):
    total_repos: int
    followers: int
    following: int
    total_stars: int
    total_forks: int
    languages: list[LanguageStat]
    top_repositories: list[RepositorySummary]


class AnalysisResponse(BaseModel):
    username: str
    github_profile: GitHubProfile
    repositories: list[RepositorySummary]
    statistics: Statistics
    roast: str
    created_at: datetime


class HealthResponse(BaseModel):
    status: str = "healthy"
